apiserver/facades/client/client/status.go

```diff
@@ -621,19 +639,30 @@ func fetchAllApplicationsAndUnits(
 		allBindingsByApp[bindings.AppName] = bindings.Bindings
 	}
 
+	charmProfileNames := make(map[charm.URL]lxdProfileName)
 	for _, app := range applications {
 		appMap[app.Name()] = app
 		appUnits := allUnitsByApp[app.Name()]
+		charmURL, _ := app.CharmURL()
+
 		if len(appUnits) > 0 {
 			unitMap[app.Name()] = appUnits
 			// Record the base URL for the application's charm so that
 			// the latest store revision can be looked up.
-			charmURL, _ := app.CharmURL()
 			if charmURL.Schema == "cs" {
 				latestCharms[*charmURL.WithRevision(-1)] = nil
 			}
 		}
+
+		// Create some lazy mapping between charm urls and serialised lxd
+		// profile names
+		charmProfileNames[*charmURL] = lxdProfileName{
```

Comment:
The machines instanceData should have a slide of profile names used by the individual machines. machine.CharmProfiles()